pkg/core/statesync/neotest_test.go

```diff
@@ -22,262 +22,413 @@ func TestStateSyncModule_Init(t *testing.T) {
 	)
 	spoutCfg := func(c *config.Blockchain) {
 		c.StateRootInHeader = true
-		c.P2PStateExchangeExtensions = true
 		c.StateSyncInterval = stateSyncInterval
 		c.MaxTraceableBlocks = maxTraceable
 	}
-	bcSpout, validators, committee := chain.NewMultiWithCustomConfig(t, spoutCfg)
-	e := neotest.NewExecutor(t, bcSpout, validators, committee)
-	for range 2*stateSyncInterval + int(maxTraceable) + 2 {
-		e.AddNewBlock(t)
-	}
 
 	boltCfg := func(c *config.Blockchain) {
 		spoutCfg(c)
+		c.P2PStateExchangeExtensions = true
 		c.Ledger.KeepOnlyLatestState = true
 		c.Ledger.RemoveUntraceableBlocks = true
 	}
 
-	t.Run("inactive: spout chain is too low to start state sync process", func(t *testing.T) {
-		bcBolt, _, _ := chain.NewMultiWithCustomConfig(t, boltCfg)
-		module := bcBolt.GetStateSyncModule()
-		require.NoError(t, module.Init(uint32(2*stateSyncInterval-1)))
-		require.False(t, module.IsActive())
-	})
+	boltCfgStorage := func(c *config.Blockchain) {
+		spoutCfg(c)
+		c.Ledger.KeepOnlyLatestState = true
+		c.Ledger.RemoveUntraceableBlocks = true
+		c.NeoFSStateSyncExtensions = true
+		c.NeoFSStateFetcher.Enabled = true
+		c.NeoFSBlockFetcher.Enabled = true
+	}
 
-	t.Run("inactive: bolt chain height is close enough to spout chain height", func(t *testing.T) {
+	t.Run("pre-init: MPT-based AddMPTNodes error", func(t *testing.T) {
 		bcBolt, _, _ := chain.NewMultiWithCustomConfig(t, boltCfg)
-		for i := uint32(1); i < bcSpout.BlockHeight()-stateSyncInterval; i++ {
-			b, err := bcSpout.GetBlock(bcSpout.GetHeaderHash(i))
-			require.NoError(t, err)
-			require.NoError(t, bcBolt.AddBlock(b))
-		}
-		module := bcBolt.GetStateSyncModule()
-		require.NoError(t, module.Init(bcSpout.BlockHeight()))
-		require.False(t, module.IsActive())
-	})
-
-	t.Run("error: bolt chain is too low to start state sync process", func(t *testing.T) {
-		bcBolt, validatorsBolt, committeeBolt := chain.NewMultiWithCustomConfig(t, boltCfg)
-		eBolt := neotest.NewExecutor(t, bcBolt, validatorsBolt, committeeBolt)
-		eBolt.AddNewBlock(t)
-
 		module := bcBolt.GetStateSyncModule()
-		require.Error(t, module.Init(uint32(3*stateSyncInterval)))
+		require.Error(t, module.AddMPTNodes([][]byte{}))
 	})
-
-	t.Run("initialized: no previous state sync point", func(t *testing.T) {
+	t.Run("pre-init: MPT-based AddContractStorageItems panic", func(t *testing.T) {
```

Comment:
No need to duplicate tests. This functionality is already covered in TestStateSyncModule_RestoreBasicChain.